public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,387 @@
+  **WHERE SHOULD I LINK TO? THE API REF DOESN'T GO INTO THESE METHODS**
+
+  You can inspect an individual `FormControl` in the demo, such as the one for Name, by using 
+  the `.get()` method. If you're curious, add the following to the template after the `form.value | json` 
+  interpolation. It isn't necessary for this demo.
+
++makeExample('reactive-forms/ts/app/hero-signup-4.component.html', 'inspect-value','app/hero-signup.component.html')(format=".")
+
+:marked
+  To get data on a `FormControl` that’s in a `FormGroup`, use the same technique, but with dot notation.
+
++makeExample('reactive-forms/ts/app/hero-signup-4.component.html', 'inspect-child-control','app/hero-signup.component.html')(format=".")
+
+:marked
+  One reason for inspecting the properties of the form is to make sure users fill it out properly. 
+  To do that, the app needs validations. To read more about validating Angular forms, see 
+  [Form Validation](../cookbook/form-validation.html). Ok, back to the demo.
+
+
+.l-main-section
+a#set-data
+:marked
+  ## Setting data with patchValue and setValue
+  You can set default data in the form using `patchValue` or `setValue`. The main difference between 
+  the two is that `patchValue` allows you to provide values as needed, specifying only the fields you 
+  want to provide values for where `setValue` requires that you pass in every control in the form.
+
+  Another important difference is that `setValue` performs strict checks and if there’s a missing 
+  control or the arguments don’t match the group structure, it returns helpful errors. On the other 
+  hand, `patchValue` doesn’t return any errors.
+
+  To use `patchValue`, add it to the constructor in the component class and pass in an object of key value 
+  pairs for the **FormControls** you want to give default values. Here, the argument is an object that provides 
+  a value for the name `FormControl`.
+
++makeExample('reactive-forms/ts/app/hero-signup-versions.component.ts', 'v5','app/hero-signup.component.ts (excerpt)')(format=".")
+
+:marked
+  Here is the same approach but using `setValue`. Notice that you have to mirror the structure of the form 
+  model and provide a value for every single `FormControl`.
+
+
++makeExample('reactive-forms/ts/app/hero-signup-versions.component.ts', 'set-value','app/hero-signup.component.ts (excerpt)')(format=".")
+
+:marked
+  With `setValue`, you can give it an empty value as in the password and confirm **FormControls**.
+
+
+
+.l-main-section
+a#form-array
+:marked
+  ## Use _FormArray_ to add groups dynamically
+  Say you want to give the user the option to add another address, and another, and another, ad infinitum. 
+  This is the perfect task for `FormArray`. You imported it in `hero-signup.component.ts` when creating 
+  the [reactive forms component](#create-comp) at the beginning of this guide.
+
+  You’ll need to refactor a little. In the component class, add an `addresses` property as type `FormArray`.
```

Comment:
May want to provide an overview of what a FormArray *is*. The above does a good job of defining when to use them ... but not really what they are. May also want to provide some context for how to use them before going into the detail of building them. Otherwise the reader has no context for the purpose of buildArray, buildGroup, etc.